### ISSUE TITLE
HP LaserJet 1018 connected to networkwed computer fails

### ISSUE THREAD

**Issue by che2lf** [apple/cups#4995]
Please help me through this out,
as title says HP1018 is connected on other machine in network, IP:192..., and i have been using it for while while i used windows xp. Installation was easy, as soon as I enter printers address http://192.168.0.218:631/printers/HP_LaserJet_1018 it worked as a charm. And it is every time since I installed it on other two laptops. But when I tried the similar thing with centos 7, via cups web interface it keeps returning filter failed error. There are three different drivers for this printer model and i tried them all. Changing protocols (http, ipp, https..) brings no changes in outcome. I am feeling like in the dead end.

**Comment by jsmeix** [comment]
This is neither a CUPS nor a Linux issue.
The root cause is too weak 'printer' hardware
(actually that thingy is not a real printer).

When a HP LaserJet 1018 is powered on
it is not yet a printer because first and foremost
it needs firmware upload to be usable for printing.

After it got its firmware it is still no normal printer
because it needs a special driver from HP's
Linux driver project HPLIP at
http://hplipopensource.com/hplip-web/index.html

In particular regarding your model see
http://hplipopensource.com/hplip-web/models/laserjet/hp_laserjet_1018.html
and note therein
<pre>
This printer REQUIRES a downloadable driver plug-in,
which is required to enable print, fax or scan support.
Use hp-setup to install the printer, and to download
and install the plug-in. Driver plug-ins are released
under a proprietary (non-open) license and are
not part of the HPLIP tarball release.
</pre>
and
<pre>
Network I/O        No
...
Network support indicates built-in ethernet and/or
wireless networking. Alternatively, many devices
may be operated on the network using an external
JetDirect print server. Not all network configurations
are supported. Please refer to the HPLIP FAQs
for more information.
</pre>

Simply put this says that you cannot use that
thingy when it is connected via network.

Buy a real printer if you want to print
instead of playing around, cf.
https://en.opensuse.org/SDB:Printer_buying_guide

For my own experience with that kind of devices see
https://bugs.launchpad.net/hplip/+bug/187049

**Comment by michaelrsweet** [comment]
Given this is a driver issue, closing.

**Comment by che2lf** [comment]
Well, nice to hear that, but still it can print from other devices. It is not a network printer, rather it is USB connected to networked ubuntu server. Most of info you could figure out from my previous post. I am sorry for being too overal. 
Of course i installed hplip, but when I run hp-setup it can not see the printer, "no device found on bus: net". hp-plugin goes well, it installes something and passes. I can reinstall still ok. hp-doctor passes, and hp-check with no errors.
You still can sound very smart putting me off and that equals all you smart linux gurus, but you can put on some idea to work this bug out. My suggestion is that something is overseen. cups can determine if printer is turned off, sometimes, and it works on remote suse ubuntu server and every windows tested. It can work from here deffinitly.
